operator/pkg/ciliumendpointbatch/utils.go

```diff
@@ -0,0 +1,73 @@
+// Copyright 2021 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ciliumendpointbatch
+
+import "github.com/cilium/cilium/pkg/lock"
+
+type CEPName string
+type CEBName string
+
+// cepToCebMapping is used to map CiliumEndpoint name to CiliumEndpointBatch name.
+// This is used by CEB manager, for every CEP Insertion/Removal CEB Controller
+// use this map to find in which CEB the CEP is queued.
+// This map is protected by lock for consistent and concurrent access.
+type cepToCebMapping struct {
```

Comment:
This whole thing seems like a really generic thread-safe map, can we using some existing, well-established lib for this? like `sync.Map`?

Reply:
At this moment i don't see any need for this `sync.Map`. Existing map with lock is suffice.

Reply:
I'm saying this whole `cepToCebMapping` should be replaced by `sync.Map`.

Reply:
As i mentioned earlier, i don't see any reason to go with `sync.Map` 
Here is the quote  from `Map` [doc](url)
..._The Map type is specialized. Most code should use a plain Go map instead, with separate locking or coordination, for better type safety and to make it easier to maintain other invariants along with the map content._